extensions/amp-story/1.0/prerender-active-page.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {escapeCssSelectorIdent} from '../../../src/css';
+import {parseQueryString} from '../../../src/url';
+import {toWin} from '../../../src/types';
+
+/**
+ * Returns true if the page should be prerendered (for being an active page or
+ * first page).
+ * @param {!AmpElement} pageElement
+ * @return {boolean}
+ */
+export function isPrerenderActivePage(pageElement) {
+  const win = toWin(pageElement.ownerDocument.defaultView);
+  const hashId = parseQueryString(win.location.href)['page'];
+  let selector = 'amp-story-page:first-of-type';
+  if (hashId) {
+    selector += `, amp-story-page#${escapeCssSelectorIdent(hashId)}`;
+  }
+  const selectorNodes = win.document.querySelectorAll(selector);
+  return selectorNodes[selectorNodes.length - 1] === pageElement;
```

Comment:
The issue with this solution is that if a publisher provides a `hashId` that doesn't match any page ID, it won't fallback on the first page.
I think this is fine if you think the performance improvement is in any way "significant". If we're talking 0.0001ms we should keep the existing implementation.

cc @mszylkowski who worked on it, FYI

Reply:
I see... To avoid making an expensive querySelector multiple times, we could cache the result on a global variable and reference that without compromising much on performance right? Only the first call to this method would be somewhat slower then

Reply:
We don't need to cache for this API. IMHO, you can make this change. But I'd rather not have it for this PR, since I'm trying to keep it to a clean refactoring.